### PR TITLE
feat(editor): Open template credential setup from collection

### PR DESCRIPTION
**Pull request by tomi**
Open the template credential setup when using a template from the collection view.

NOTE! This feature is still behind a feature flag. To test, set:

```js
localStorage.setItem('template-credentials-setup', 'true')
```

https://github.com/n8n-io/n8n/assets/10324676/46ccec7c-5a44-429e-99ad-1c10640e99e5

### CODE REVIEW

File: cypress/e2e/34-template-credentials-setup.cy.ts [hunk 1]
```diff
@@ -28,6 +31,18 @@ describe('Template credentials setup', () => {
 			.should('be.visible');
 	});
 
+	it.only('can be opened from template collection page', () => {
+		templateCollectionPage.actions.visit(COLLECTION_ID);
```
Comment (mutdmour) [inline cypress/e2e/34-template-credentials-setup.cy.ts:35]: should not we also mock the collection api request?

File: cypress/pages/template-collection.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { BasePage } from './base';
+
+export class TemplateCollectionPage extends BasePage {
```
Comment (mutdmour) [inline cypress/pages/template-collection.ts:3]: we are trying to move away from this pattern.. checkout `pages/demo.ts` for an example

File: packages/editor-ui/src/utils/templates/templateActions.ts [hunk 3]
```diff
@@ -35,3 +38,32 @@ export async function createWorkflowFromTemplate(
...
+		? {
+				name: VIEWS.TEMPLATE_SETUP,
+				params: { id: templateId },
+		  }
+		: {
+				name: VIEWS.TEMPLATE_IMPORT,
+				params: { id: templateId },
+		  };
+
+	if (inNewBrowserTab) {
```
Comment (mutdmour) [inline packages/editor-ui/src/utils/templates/templateActions.ts:63]: 😻

**Comment by tomi** [comment]
@mutdmour comments addressed, ready for re-review

**Review by mutdmour — APPROVED** [review]

**Pull request merged**